src/test/java/org/rulez/demokracia/pdengine/integrationtest/VoteCreationIntegrationTest.java

```diff
@@ -70,14 +72,44 @@ public void vote_creation_fails_with_404_on_bad_input() {
 	@TestedBehaviour("Creates a vote")
 	@Test
 	public void vote_creation_fails_and_reports_error_message_on_bad_input() {
-		req.voteName= "`drop table little_bobby tables;`";
-		Invocation.Builder invocationBuilder = createWebClient();
-		Response response = invocationBuilder.post(Entity.entity(req,MediaType.APPLICATION_JSON));
+		Response response = RestCallWithBadInput();
 		String responseString = response.readEntity(String.class);
-		JSONObject responseJson = new JSONObject(responseString);
+		JsonObject responseJson = new JsonParser().parse(responseString).getAsJsonObject();
 		assertEquals("invalid characters in {0}", responseJson
-				.getJSONObject("error")
-				.getString("message"));
+				.get("error").getAsJsonObject()
+				.get("message").getAsString());
+	}
+
+	@TestedFeature("Manage votes")
+	@TestedOperation("create vote")
+	@TestedBehaviour("Creates a vote")
+	@Test
+	public void vote_creation_fails_and_reports_error_details_on_bad_input() {
+		Response response = RestCallWithBadInput();
+		String responseString = response.readEntity(String.class);
+		JsonObject responseJson = new JsonParser().parse(responseString).getAsJsonObject();
+		assertEquals("vote name", responseJson
+				.get("error").getAsJsonObject()
+				.get("details").getAsString());
+	}
+
+	@TestedFeature("Manage votes")
+	@TestedOperation("create vote")
+	@TestedBehaviour("Creates a vote")
+	@Test
+	public void vote_creation_fails_and_returns_the_bad_input() {
+		Response response = RestCallWithBadInput();
+		String responseString = response.readEntity(String.class);
+		JsonObject responseJson = new JsonParser().parse(responseString).getAsJsonObject();
+		assertEquals(new Gson().toJson(req), responseJson
+				.get("input").toString());
+	}
+
+	private Response RestCallWithBadInput() {
```

Comment:
naming conventions

Reply:
That's why style check would be useful.

Reply:
See the video about how to use PMD at home. PMD spots this.